OutSystems/outsystems-maps
# ROU-11055: Adding Google Advanced Markers and Map Style Id properties

This PR is for adding Google Advanced Markers and the ability to use styles created in [Google Style Console](https://console.cloud.google.com/google/maps-apis/studio/styles).

### What was happening
With version `3.56`, google deprecated the class `google.maps.Marker` with the following message*:
```
As of February 21st, 2024, google.maps.Marker is deprecated. Please use google.maps.marker.AdvancedMarkerElement instead. At this time, google.maps.Marker is not scheduled to be discontinued, but google.maps.marker.AdvancedMarkerElement is recommended over google.maps.Marker. While google.maps.Marker will continue to receive bug fixes for any major regressions, existing bugs in google.maps.Marker will not be addressed. At least 12 months notice will be given before support is discontinued. Please see https://developers.google.com/maps/deprecations for additional details and https://developers.google.com/maps/documentation/javascript/advanced-markers/migration for the migration guide.
```

Additionally, when using the new `AdvancedMarkerElement`, it become necessary to use the attribute `mapId`, which is used to indicate the Map Style Id for the styles created in Google Style Console. However, by using it, the current styles stop working, and a new warning is displayed in the console**:
```Google Maps JavaScript API: A Map's styles property cannot be set when a mapId is present. 
When a mapId is present, Map styles are controlled via the cloud console. 
Please see documentation at https://developers.google.com/maps/documentation/javascript/styling#cloud_tooling
```

*Consider WARNING1
**Consider WARNING2

### What was done
* Added a new configuration (`UseAdvancedMarker`) to allow the developer to define which version of the Marker should be used.
* Added a new configuration (`MapStyleId`) to allow the developer to pass the style Id as defined in the cloud console:
![image](https://github.com/user-attachments/assets/c38753a1-f674-4394-9462-613642745903)

* Updated the version of the google maps being used from `3.55` to `3.57` (see [version list](https://developers.google.com/maps/documentation/javascript/versions))
* Updated the version of OutSystems Maps to `2.0.0`


### Test Steps
#### Test 1 - happy path
1. Enter the test page
2. And set the following configs for Map 1:
   a. Use Advance Markers: true
   b. Deprecated Style: Standard
3. Set Show Map (1): true
4. Result:
  - a map should be displayed with styles coming from the google cloud console
  - no warning in the developer tools

#### Test 2
1. Enter the test page
2. And set the following configs for Map 1:
   a. Use Advance Markers: true
   b. Deprecated Style: Retro
3. Set Show Map (1): true
4. Result: 
  - a map should be displayed with styles coming from the google cloud console
  - a warning will be displayed in the developer tools:  `WARNING1`

#### Test 3
1. Enter the test page
2. And set the following configs for Map 1:
   a. Use Advance Markers: false
   b. Deprecated Style: Retro
3. Set Show Map (1): true
4. Result:
  - a map should be displayed with the Retro styles
  - a warning will be displayed in the developer tools: `WARNING2`
  
#### Test 4
1. Enter the test page
2. And set the following configs for Map 1:
   a. Use Advance Markers: true
   b. Deprecated Style: Standard
3. And set the following configs for Map 2:
   a. Use Advance Markers: false
   b. Deprecated Style: Standard
4. Set Show Map (1): true
5. Set Show Map (2): true
6. Result:
   - Both maps will be displayed
   - Map 1 with google cloud console styles
   - Map 2 with standard styles
   - a warning will be displayed in the developer tools: `WARNING1`

### Checklist
